test/IntegrationTests/FileVectorizer/test_deletefiles.py

```diff
@@ -0,0 +1,363 @@
+            start_time = time.time()
+
+            while True:
+                response = requests.get(
+                    embedding_status_url,
+                    headers=customer["headers"],
+                    params={"page": 1, "page_size": 100}
+                )
+                self.assertEqual(response.status_code, 200)
+
+                relevant_status = [
+                    f for f in response.json()
+                    if f["filename"] in dict(customer["uploaded_files"])
+                ]
+
+                logger.info(f"Customer {idx + 1} - {len(relevant_status)}/{len(customer['uploaded_files'])} files:")
+                for file_status in relevant_status:
+                    logger.info(
+                        f"file_id: {file_status.get('fileid')} | "
+                        f"filename: {file_status.get('filename')} | "
+                        f"embeddingstatus: {file_status.get('embeddingstatus')}"
+                    )
+
+                if all(f["embeddingstatus"] == "SUCCESS" for f in relevant_status):
+                    logger.info(f"Customer {idx + 1}: All files embedded in {time.time() - start_time:.1f}s!")
+                    break
+
+                time.sleep(3)
+
+        logger.info("==== Deleting Files + Vectorizer Control ====")
+        for idx, customer in enumerate(customers):
+            logger.info(f"Customer {idx + 1}: Initiating deletion...")
+
+            # First collect all filenames for this customer
+            customer_filenames = {filename for filename, _ in customer["uploaded_files"]}
+
+            # Delete all files first
+            for filename, file_id in customer["uploaded_files"]:
+                response = requests.delete(
+                    delete_url,
+                    params={"filename": filename},
+                    headers=customer["headers"]
+                )
+                self.assertEqual(response.status_code, 200, f"❌ Failed to delete {filename}")
+                logger.info(f"Initiated deletion for {filename}")
+
+            # Then monitor status
+            last_control_time = time.time()
+            control_interval = 30
+            start_time = time.time()
+            deleted_files = set()
+
+            while True:
+                try:
+                    current_time = time.time()
+
+                    # Vectorizer control (if needed)
+                    if current_time - last_control_time >= control_interval:
+                        logger.info("Stopping file vectorizer...")
+                        self.control_file_vectorizer()
+                        last_control_time = current_time
+
+                    # Check deletion status
+                    response = requests.get(
+                        deletion_status_url,
+                        headers=customer["headers"],
+                        params={"page": 1, "page_size": 100}
+                    )
+                    self.assertEqual(response.status_code, 200)
+
```

Comment:
Add an assert statement to check whether the number of files initiated for deletion is 10